orttraining/orttraining/core/optimizer/compute_optimizer/padding_elimination.cc

```diff
@@ -361,6 +369,30 @@ void IterateSubgraphFromNode(Graph& graph,
       } else {
         candidate_outputs.insert(cur);
       }
+    } else if (graph_utils::IsSupportedOptypeVersionAndDomain(*cur, "ReduceMean", {1, 11, 13, 18})) {
+      if (cur->InputDefs()[0]->Shape()) {
+        auto axes = cur->GetAttributes().at("axes").ints();
+        bool axes_check = (axes.size() > 0);
+        for (int64_t axis : axes) {
+          axis = axis < 0 ? axis + cur->InputDefs()[0]->Shape()->dim_size() : axis;
+          if (axis < 2) {
+            LOG_DEBUG_INFO(logger, "PaddingElimination::axis of ReduceMean: " + cur->Name() + " is " +
+                                       std::to_string(axis) + ", which blocks merging leading two dims.");
+            candidate_outputs.insert(cur);
+            axes_check = false;
+            break;
+          }
+        }
+        if (axes_check) {
+          LOG_DEBUG_INFO(logger, "PaddingElimination::ReduceMean: " + cur->Name() + " is added to subgraph.");
+          subgraph.insert(cur->MutableOutputDefs()[0]);
+          PushAllOutputNode(graph, to_visit, cur, visited);
+        }
```

Comment:
for axes_check being false, we need also add cur in candidate_outputs?

Reply:
Yep! you are right! thanks, done.